packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -747,16 +747,31 @@ export class UserPool extends Resource implements IUserPool {
         }),
         inlinePolicies: {
           /*
-           * The UserPool is very particular that it must contain an 'sns:Publish' action as an inline policy.
-           * Ideally, a conditional that restricts this action to 'sms' protocol needs to be attached, but the UserPool deployment fails validation.
-           * Seems like a case of being excessively strict.
+           * The Cognito service is very particular that 'sns:Publish' action must be on an inline policy.
+           * Ideally, a conditional that restricts this action to 'sms' protocol needs to be attached, but SNS doesn't
+           * support this yet. So allow all first and then restrict publishing to topics and emails.
            */
           'sns-publish': new PolicyDocument({
             statements: [
               new PolicyStatement({
                 actions: [ 'sns:Publish' ],
                 resources: [ '*' ],
-              })
+              }),
+              new PolicyStatement({
+                actions: [ 'sns:Publish' ],
+                effect: Effect.DENY,
+                resources: [ Stack.of(this).formatArn(
+                  { partition: '*', region: '*', account: '*', service: 'sns', resource: '*', } // all topics
+                ) ],
+              }),
+              new PolicyStatement({
+                actions: [ 'sns:Publish' ],
+                effect: Effect.DENY,
+                resources: [ '*' ],
+                conditions: {
+                  StringEquals: { 'sns:Protocol': 'email' }
```

Comment:
Since SNS does not support sending email directly, but only through a topic (afaict, am I wrong?), this one seems unnecessary since it would be covered by the previous statement?

Reply:
I think you are correct here. This does nothing.